package/piwebcam/multi-gadget.sh

```diff
@@ -64,17 +69,20 @@ EOF
 config_usb_webcam () {
   mkdir -p functions/uvc.usb0/control/header/h
 
-  config_frame mjpeg m  640  360
-  config_frame mjpeg m  640  480
-  config_frame mjpeg m  800  600
-  config_frame mjpeg m 1024  768
-  config_frame mjpeg m 1280  720
-  config_frame mjpeg m 1280  960
-  config_frame mjpeg m 1440 1080
-  config_frame mjpeg m 1536  864
-  config_frame mjpeg m 1600  900
-  config_frame mjpeg m 1600 1200
-  config_frame mjpeg m 1920 1080
+  if [ ! -e $VIDEO_FORMATS_USER_FILE ] ; then
```

Comment:
This sounds the wrong way round. If user file does NOT exist, use user file as formats file? Did you check this with a custom file that does specify some other formats?

Also, how do you feel about a stronger check, i.e. "file is readable" `-r` or even "not empty" `-s`

Reply:
Right, you could set ` VIDEO_FORMATS_FILE=/etc/video_formats.txt` as default, and overwrite it if a user file exists. It should always be readable, as /boot is FAT. But you're right, better safe than sorry :)